internal/controller/operatorconfigmap_controller.go

```diff
@@ -532,9 +558,20 @@ func (c *OperatorConfigMapReconciler) getNoobaaSubManagementConfig() bool {
 	return val
 }
 
-func (c *OperatorConfigMapReconciler) shouldGenerateRBDOmapInfo() bool {
-	valAsString := strings.ToLower(c.operatorConfigMap.Data[generateRbdOMapInfoKey])
-	return valAsString == strconv.FormatBool(true)
+func (c *OperatorConfigMapReconciler) shouldGenerateOmapInfo() (bool, error) {
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := c.list(storageClients); err != nil {
+		return false, err
+	}
+
+	for idx := range storageClients.Items {
+		if storageClients.Items[idx].Status.MirrorEnabled {
+			return true, nil
+		}
+
+	}
```

Comment:
If we are to generate omap data in cases where a single client requests it, how would that affect the other clients that don't need it?

Reply:
If any client is requesting for the omap info, we are going to enable it, as this is a operator level operation we cannot restrict it to per client

Reply:
I understand, but I was asking about the impact/the side effects. Not the procedure

Reply:
The impact would be that we enable it for all other clients as well. I don't think we have any side effects if it is always enabled